module-2/backend/week_two.md

```diff
@@ -4,38 +4,89 @@ Fork this respository. Answer the questions to the best of your ability. Try to
-13. `Rack::Test` allows us to test our controllers in isolation. What are some of the methods it gives us to simulate the request/response cycle?
-14. Describe some techniques to make our Sinatra views more DRY. Give an example of when you would use these techniques.
+1. At a high level, what is ActiveRecord? What does it do/allow you to do?  
+ActiveRecord is an ORM for using Ruby with relational SQL databases. 
+ActiveRecord is a collection of methods written to streamline common Ruby methods. 
+
+2. What is a migration?  
+A migration is the instruction for how to alter/create tables in a database.   
+
+3. How does a table relate to a model?  
+A model represents a table in a database. You can build methods on a model to interact with a table.   
+
+4. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.  
+They are class methods. Class and Teacher would have a one to many relationship, a class has one teacher and teachers have many classes. Students and Class would have a many to many relationships as a class has many students and students have many classes.   
+
+5. What do they allow you to do?  
+These associations allow tables to have relationships with other tables and the information within them to be accessible through their related tables.   
+
+6. What's the difference between agile workflow and waterfall method?  
+Agile workflow involves breaking apart a project into small chunks, completing all aspects of that chunk before moving on to another one. Waterfall moves through each stage of developmentin sequence for the whole project.  
+
+7. What is the difference between `#new` and `#create`?  
+New makes a new instance, create makes a new instance and saves it.   
+
+8. At a basic level, what does cURL allow you to do?  
+Pull in url from the command line.    
+
+9. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.  
+Students and Teachers have a many to many relationship. Stu have many Tch and Tch have many stu.  They would be connected in a db through a join table i.e. students_teachers  
+
+10. Define foreign key, primary key, and schema.  
+Foreign key is a key that relates one table to another.  
+Primary key is the key that is a unique identifier on a table.  
+Schema is the overall structure of the database - the tables, what is in them, and how they relate to each other.   
+
+11. Describe the relationship between a foreign key on one table and a primary key on another table.  
+A primary key become the foreign key on another table. For example if you had class and teacher (one to many), the class would have a foreign key which is comprised of teacher's primary key.   
+
+12. What are the parts of an HTTP response?  
+Status, Header, Body   
+
+13. `Rack::Test` allows us to test our controllers in isolation. What are some of the methods it gives us to simulate the request/response cycle?  
```

Comment:
Think we removed this from the latest version. We didn't teach Rack::Test this time around, but you'll see controller testing in Mod 3.